core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -1002,4 +1000,12 @@ private object ApiHelper {
     }
   }
 
+  def lastStageNameAndDescription(store: AppStatusStore, job: JobData): (String, String) = {
+    store.asOption(store.lastStageAttempt(job.stageIds.max)) match {
+      case Some(lastStageAttempt) =>
+        (lastStageAttempt.name, lastStageAttempt.description.getOrElse(job.name))
+      case None => ("", "")
```

Comment:
Before, you were doing `if (lastStageDescription.isEmpty) job.name else blah` at the call site.

Now, when the last stage is not in the store, the call site is getting an empty string as the description, instead of using the job name.

Reply:
This would probably be simpler:

```
val stage = store.asOption(...)
(stage.map(_.name).getOrElse(""), stage.map(_.description.getOrElse(job.name)))
```

Reply:
Fixed, thanks for catching.